[FIX] Dockerfile: add env var to avoid numpy weird errors

fernandahf opened the pull request:
Description of the issue/feature this PR addresses:

Currently, if numpy is available in the modules even if you are not using it
Odoo try to compile and the system is down only for a type of processor

Currently we know 2 server reproducing the error:

- B&F-production
- Runbot

More info about:

https://github.com/numpy/numpy/issues/17674
https://github.com/numpy/numpy/pull/17759

It is reproducing in the following MR:

https://git.vauxoo.com/vauxoo/lasec/-/merge_requests/197

Check the following discussion https://odoo-community.org/groups/contributors-15/contributors-186006?mode=thread&date_begin=&date_end=

OpenBLAS creates a number of threads equal to the number of core threads available: 56 in my case (production server),
so it quickly reached limit_memory_hard
and the process was killed (SIGSEGV)
Forcing OPENBLAS_NUM_THREADS=1 fixed the issue.

Current behavior before PR:
Weird errors related to numpy.

Desired behavior after PR is merged:
No errors related to numpy.

Close #85

-- I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

fernandahf committed:
[FIX] Dockerfile: add env var to avoid numpy weird errors

ruiztulio commented:
@fernandahf Can you check why the pipeline is showing an error? this error in in the 16.04-p3 image (the one you are making the PR to) I know is not related to your change, but the change won't land in production if we cannot build the image properly

\cc @moylop260

fernandahf committed:
[REF] scripts: update py_download_execute and URLs of get-pip

Default python version for pip is 3.6, other python versions
should be use other URLs, otherwise an error is raised:

ERROR: This script does not work on Python 2.7 The minimum supported
Python version is 3.6. Please use https://bootstrap.pypa.io/pip/2.7/get-pip.py instead.

fernandahf commented:
For record: 

There is an error with cryptography package with images with python versions < 3.6:

https://travis-ci.org/github/Vauxoo/docker-ubuntu-base/builds/767209867#L1049-L1064

We working with Odoo version >= 12.0, therefore, it's not neccesary update this repo.

I will close it.
